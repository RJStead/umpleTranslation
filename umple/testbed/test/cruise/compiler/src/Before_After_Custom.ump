generate Java "../src-gen-umple";

class BeforeAfterCustom
{
  String foo(int a) {

    if(a == 3) {
        return "3";
    }
    
    a = a*2;
    
    while (a > 0) {
        a--;
        if (a % 10 == 0) {
            a = a / 10;
            return a;
        }
    }
    
    System.out.println("About to exit");
    return "";
  }

  before custom foo(int) {
    System.out.println("Starting foo...");
  }

  after custom foo(int) {
    String b = "1";
    System.out.println("Returning from foo, a: " + String.valueOf(a) + ", b: " + b);
  }

  before custom foo(int) {
    // This is a comment
    int x = 1;
    a += x;
  }
}
