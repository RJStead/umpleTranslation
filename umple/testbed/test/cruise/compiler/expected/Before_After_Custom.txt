Before_After_Custom.ump:17: error: incompatible types: int cannot be converted to String
            return a;
                   ^
1 error